src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/Controller/ControllerGeneratorBase.cs

```diff
@@ -83,13 +84,31 @@ protected virtual IEnumerable<string> TemplateFolders
             }
         }
 
-        protected string GetControllerNamespace()
+        protected string GetDefaultControllerNamespace()
         {
             // Review: MVC scaffolding used ActiveProject's MSBuild RootNamespace property
             // That's not possible in command line scaffolding - the closest we can get is
             // the name of assembly??
             var appName = ApplicationInfo.ApplicationName;
-            return appName + "." + Constants.ControllersFolderName;
+
+            var defaultControllerNamespace = appName + "." + Constants.ControllersFolderName;
```

Comment:
Issue 229 complained about Controllers instead of ApiControllers. Why don't we look at current folder name and take it first as default namespace?